extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusPermission.java

```diff
@@ -0,0 +1,147 @@
+package io.quarkus.security.runtime;
+
+import java.security.Permission;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.security.PermissionChecker;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.spi.runtime.BlockingSecurityExecutor;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Special type of the {@link PermissionsAllowed#permission()} that does not require {@link SecurityIdentity}
```

Comment:
IMHO this is too implementation detail specific where you imply users can actually use it, and then, later, say users should not use it.
Can we just say `Special type of 'Permission' which is used by Quarkus Security to call CDI bean methods annotated with '{@link PermissionsChecker}' ` ? IMHO users don't need to know at the moment how this permission works internally

Reply:
Michal, @michalvavrik, I hope you will be around for a long while :-), so you will explain if someone will want clarifications :-).